File: 1.Fundamental-Level/1.Advanced-C#/02.Multidimensional-Arrays-Sets-Dictionaries/06.CountSymbols/CountSymbols.cs
namespace _06.CountSymbols
{
    using System;
    using System.Collections.Generic;

    public class CountSymbols
    {
        public static void Main()
        {
            var symbolsCount = new SortedDictionary<char, int>();

            string text = Console.ReadLine();
            foreach (char symbol in text)
            {
                if (!symbolsCount.ContainsKey(symbol))
                {
                    symbolsCount.Add(symbol, 1);
                }
                else
                {
                    symbolsCount[symbol]++;
                }
            }

            foreach (var pair in symbolsCount)
            {
                Console.WriteLine("{0}: {1} time/s", pair.Key, pair.Value);
            }
        }
    }
}

File: Java-Development/Advanced-Java/02.Sets-and-Maps-Exercises/src/_04_CountSymbols/CountSymbols.java
package _04_CountSymbols;

import java.util.Map;
import java.util.Scanner;
import java.util.TreeMap;

public class CountSymbols {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String text = scanner.nextLine();

        TreeMap<Character, Integer> symbolsCount = new TreeMap<>();
        for (int i = 0; i < text.length(); i++) {
            Character symbol = text.charAt(i);
            if (!symbolsCount.containsKey(symbol)) {
                symbolsCount.put(symbol, 0);
            }

            int count = symbolsCount.get(symbol) + 1;
            symbolsCount.put(symbol, count);
        }

        for (Map.Entry<Character, Integer> symbol : symbolsCount.entrySet()) {
            System.out.printf("%s: %d time/s%n", symbol.getKey(), symbol.getValue());
        }
    }
}
